tests/php/core-api/test_classs.jetpack-core-api-authentication.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+require_once JETPACK__PLUGIN_DIR . '/tests/php/lib/class-wp-test-rest-controller-testcase.php';
+require_once JETPACK__PLUGIN_DIR . '/tests/php/lib/class-wp-test-spy-rest-server.php';
+
+class WP_Test_Jetpack_Core_Api_Authentication extends WP_Test_REST_Controller_Testcase {
+
+    /**
+     * @author rooctripaldi
+     * @covers Jetpack->wp_json_authenticate
+     * @requires PHP 5.2
+     */
+    public function test_jetpack_server_authorization_fail() {
+        $request = new WP_REST_Request( 'GET', '/jetpack/v4/module/all' );
+        $request->set_param( 'token', 'an invalid token' );
+        $this->server->check_authentication();
+        $response = $this->server->dispatch( $request );
+        $this->assertEquals( 403, $response->get_status() );
+    }
```

Comment:
This is a good start, but I think it's really important to get full branch coverage on the `wp_json_authenticate` (or `wp_rest_authenticate`) method and all of its possible return values:

- failed `verify_xml_rpc_signature`
- auth succeeded
- generic `rest_forbidden` failure message

The error tests should include verification that the correct error is returned, not just that an error was returned.  The easiest way to do that is to return a different error code for each situation and test the error codes.

The WP core test suite has a nice helper for this:  [`assertErrorResponse`](https://github.com/nylen/WordPress/blob/5d832bc2aebc98bd806bb3758adb2e0fbde1517f/tests/phpunit/includes/testcase-rest-api.php#L4)